src/cim/components/ClusterDeployment/helpers.ts

```diff
@@ -92,6 +92,9 @@ export const isCIMFlow = (clusterDeployment?: ClusterDeploymentK8sResource) =>
     INFRAENV_GENERATED_AI_FLOW
   ];
 
+export const isAgentOfInfraEnv = (infraEnv: InfraEnvK8sResource, agent: AgentK8sResource) =>
+  agent.metadata?.labels?.[INFRAENV_AGENTINSTALL_LABEL_KEY] === infraEnv.metadata?.name;
```

Comment:
we should also check that the namespace is the same, to be sure

Reply:
The agent is created in the namespace of the InfraEnv and just the infraEnv's name is stored among labels. 
I can add it for future reusability of the utility function.

Reply:
done